core/src/main/scala/com/salesforce/op/OpWorkflowModelReader.scala

```diff
@@ -81,19 +82,28 @@ class OpWorkflowModelReader(val workflow: OpWorkflow) extends MLReader[OpWorkflo
    * @return workflow model instance
    */
   def loadJson(json: JValue, path: String): Try[OpWorkflowModel] = {
+
     for {
       trainParams <- OpParams.fromString((json \ TrainParameters.entryName).extract[String])
       params <- OpParams.fromString((json \ Parameters.entryName).extract[String])
       model <- Try(new OpWorkflowModel(uid = (json \ Uid.entryName).extract[String], trainParams))
       (stages, resultFeatures) <- Try(resolveFeaturesAndStages(json, path))
       blacklist <- Try(resolveBlacklist(json))
+      results <- resolveRawFeatureFilterResults(json)
       distributions <- resolveRawFeatureDistributions(json)
     } yield model
       .setStages(stages.filterNot(_.isInstanceOf[FeatureGeneratorStage[_, _]]))
       .setFeatures(resultFeatures)
       .setParameters(params)
       .setBlacklist(blacklist)
-      .setRawFeatureDistributions(distributions)
+      .setRawFeatureFilterResults(
+        if (results.rawFeatureDistributions.nonEmpty) { // for backwards compatibility
```

Comment:
please move this if/else condition to `resolveRawFeatureFilterResults` and change the return type of `resolveRawFeatureFilterResults` to `Try[RawFeatureFilterResults]`